consensus/istanbul/validator/weighted.go

```diff
@@ -705,6 +703,21 @@ func filterPoorValidators(weightedValidators []*weightedValidator, stakingAmount
 		}
 	}
 
+	// when no validator has more than minimum staking amount of KLAYs, all members are in validators
+	if len(newWeightedValidators) <= 0 {
+		// 1. if governance mode is not single,
+		// 2. if governance mode is single and governing node does not have minimum staking amount of KLAYs as well
+		if !isSingleMode || (isSingleMode && uint64(govNodeStaking) < amount) {
```

Comment:
What's the purpose of this `if` statement here?
If you want to filter out `isSingleMode && uint64(govNodeStaking) >= amount` `when len(newWeightedValidators) <= 0`, I think error case logic should be added by `else` statement.

Reply:
@ehnuje The `if` statement is for the case when all validators are demoted. Here, we need to consider two cases.
1. in `none` or `ballot` mode, (`!isSingleMode`)
  - If `len(newWeightedValidators) <= 0`, the all members are demoted. So, the demoted ones are promoted.
2. in `single` mode, (`isSingleMode`)
  - We need to consider the governing node. So, if any validator and the governing node don't have enough KLAYs, then all demoted ones become validators.
 
The current policy is that all demoted ones become validators when they do not stake enough. So, it is not necessary to add error cases. Any suggestion?

Reply:
@jeongkyun-oh I thought `govNode` will be the only validator if `len(newWeightedValidators) <= 0` and `(isSingleMode && uint64(govNodeStaking) < amount)`. However, this logic makes all validators to be listed on `newWeightedValidators`. Is that you intended?

Reply:
@aidan-kwon Yes, it is intended to make all validators to be listed. I followed the following rules in priority. I have updated PR description of how validators are selected.

In this case, if any validator including the governing node does not have minimum staking, then all validators are listed to the `newWeightedValidators`. However, if the governing node stake more than the minimum, then the one will only be the validator. What do you think?